content/200-concepts/100-components/01-prisma-schema/08-views.mdx

```diff
@@ -164,20 +164,131 @@ view UserInfo {
 
 <!-- prettier-ignore-end -->
 
-### Limitations
+### Write by Hand
+
+A `view` block is comprised of two main pieces:
+
+- The `view` block definition
+- The view's field definitions
+
+These two pieces allow you to define the name of your view in the generated Prisma Client and the columns present in your view's query results.
+
+#### Define a `view` block
+
+To define the `UserInfo` view from the example above, begin by using the `view` keyword to define a `view` block in your schema named `UserInfo`:
+
+```prisma
+view UserInfo {
+  // Fields
+}
+```
+
+<!-- It would be good to reference some recommended naming conventions as we do for `model` blocks -->
+
+#### Define fields
+
+The properties of a view are called _fields_, which consist of:
+
+- A field name
+- A field type
+
+<!-- Would be good to link off to API reference material for valid attributes on these when available -->
+
+The fields of the `UserInfo` example view can be defined as follows:
+
+<TabbedContent tabs={[<FileWithIcon text="Relational databases" icon="database"/>, <FileWithIcon text="MongoDB" icon="database"/>]}>
+<tab>
+
+```prisma highlight=2-5;normal
+view UserInfo {
+  id    Int    @unique
+  email String
+  name  String
+  bio   String
+}
+```
+
+</tab>
+<tab>
+
+```prisma highlight=2-5;normal
+view UserInfo {
+  id    String @unique
+  email String
+  name  String
+  bio   String
+}
+```
+
+</tab>
+</TabbedContent>
+
+Each _field_ of a `view` block represents a column in the query results of the view in the underlying database.
+
+### Use Introspection
+
+If you have an existing view or views defined in your database, [introspection](/concepts/components/introspection) will automatically generate `view` blocks in your Prisma schema that represents those views.
+
+Assuming the example `UserInfo` view exists in your underlying database, running the following command will generate a `view` block in your Prisma schema representing that view:
+
+```sh copy
+npx prisma db pull
+```
+
+The resulting `view` block will be defined as follows:
+
+```prisma
+view UserInfo {
+  id    Int?
+  email String?
+  name  String?
+  bio   String?
+
+  @@ignore
+}
+```
 
-Currently, Prisma treats views in the same way as models. This means that a view needs to have a [unique identifier](/concepts/components/prisma-schema/data-model#defining-an-id-field). In relational databases, this can be an `@id` or `@unique` attribute on one field, or an `@@id` or `@@unique` attribute on multiple fields. In MongoDB, this must be an `@id` attribute that maps to the `_id` field in the underlying database with `@map("_id")`.
+The `view` block is generated initially with a `@@ignore` attribute because there is no [unique identifier](/concepts/components/prisma-schema/data-model#defining-an-id-field) defined.
```

Comment:
Because the limitation about views needing a unique ID hasn't been mentioned yet, I would rather link down to that limitation instead of the docs for unique IDs.

```suggestion
The `view` block is generated initially with a `@@ignore` attribute because [there is no unique identifier defined](#unique-identifier).
```

Reply:
Or:

```suggestion
The `view` block is generated initially with a `@@ignore` attribute because there is no [unique identifier](/concepts/components/prisma-schema/data-model#defining-an-id-field) defined (which is a currently a [limitation](#unique-identifier) of the views implementation in Prisma).
```

Reply:
Actually, linking to `/concepts/components/prisma-schema/data-model#defining-an-id-field` is also not 100% ideal because instead of defining an `@id` field, it would be enough to define an `@unique` field — so ideally this gets mentioned as well.